up-l1/README.adoc

```diff
@@ -261,6 +261,38 @@ end
 deactivate T
 ----
 
+In RegisterListener API, we have the source and sink `UUri` in the arguments.
+However, not all the combinations of source and sink `UUri` are valid.
+
+The following table are the valid combinations, which maps to the corresponding user case.
+
+[%autowidth]
+|===
+| source `resource_id` | sink `resource_id` | Publish | Notification | Request | Response | Use Case
+
+| [8000-FFFE] |   None   |    V    |              |         |         | A uE listens for a specific published message 
+| [8000-FFFE] |     0    |         |      V       |         |         | A uE listens for a specific notification message
+|      0      | [1-7FFF] |         |              |    V    |         | A uE listens for a specific request 
+|   [1-7FFF]  |     0    |         |              |         |    V    | A uE listens for a specific response
+|     FFFF    |     0    |         |      V       |         |    V    | A uE listens for all notifications and responses 
+|     FFFF    |   FFFF   |         |      V       |    V    |    V    | uStreamer listens for all notifications, requests, and responses
```

Comment:
I'm not sure this is valid because it will them mean the streamer will register for listeners for itself. @PLeVasseur what are you passing to the transports now in up-streamer-rust?

Reply:
Should be okay.

This is for the resource IDs only, which is a much more narrow scope. The uStreamer still has the ability to discriminate based on the authority_name.

Here's a [snippet of code](https://github.com/eclipse-uprotocol/up-streamer-rust/blob/d4ab17fbc745705290e29e5251fa8e9fcbffe943/up-streamer/src/ustreamer.rs#L158-L159) showing how we currently use `FFFF`, `FFFF` for source and sink resource IDs. You'd have to read up a bit to see the definitions of the functions for creating the UUris [here](https://github.com/eclipse-uprotocol/up-streamer-rust/blob/d4ab17fbc745705290e29e5251fa8e9fcbffe943/up-streamer/src/ustreamer.rs#L31-L49).